frontend/hooks/useAPI.js

```diff
@@ -0,0 +1,22 @@
+import { useState } from 'react';
+import { API_URL } from "@env";
+
+const useAPI = (endpoint, onSuccess, onError) => {
```

Comment:
this custom hook accepts endpoint to send request to, onSuccess to run in case of successful response and onError to run if there is error